plugin/storage/memory/options.go

```diff
@@ -0,0 +1,40 @@
+// Copyright (c) 2018 The Jaeger Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package memory
+
+import (
+	"flag"
+
+	"github.com/spf13/viper"
+
+	"github.com/jaegertracing/jaeger/pkg/memory/config"
+)
+
+const limit = "memory.limit"
```

Comment:
This translates to a flag like `--memory.limit=100`. This is also consistent with `cassandra`, where the plugin name is the prefix for the option. What I don't like, though, is that "limit" is too generic and it's not clear what kind of limit this is. Passing `--help` will clarify it, but I'm open to suggestions for a better name. 

`--memory.max-traces` perhaps?

Reply:
Is it traces or spans?

Reply:
Traces

Reply:
I believe it's better and cleaner to bound on the total number of spans. The docs should also mention an average estimation of span size, so people roughly know what value should be used. 

I am not sure how much it would complicate your PR...

Reply:
Span size estimation is hard when you don't know what people are storing within the span... Also, evicting traces is cleaner than evicting individual spans. Otherwise, a trace might show up now in the UI with 100 spans and then, after a couple of minutes, with only 10 spans.

Reply:
> Span size estimation is hard when you don't know what people are storing within the span... 

For example it could use average HTTP span. Which is what most of the people use. 

> Also, evicting traces is cleaner than evicting individual spans

I am not saying we should evict on per span! Eviction strategy is different than limit :) 

>  Otherwise, a trace might show up now in the UI with 100 spans and then, after a couple of minutes, with only 10 spans.

This can happen with your impl too.

Reply:
> For example it could use average HTTP span. Which is what most of the people use.

I think this fits the memory-bound strategy that was mentioned before. We'll wait for #773 for that.

> I am not saying we should evict on per span! Eviction strategy is different than limit :)

OK, so, you mean to keep a count of the number of spans and evict traces when the number of spans reaches the limit? Could be a possible additional strategy and a candidate for a follow-up PR. For now, we just need a simple upper bound.